### ISSUE TITLE
docker image push failed for SAM created ecr repos in China regions

### ISSUE THREAD

**Issue by bnusunny** [aws/aws-sam-cli#3274]
### Description:

When using sam deploy to deploy a docker image packaged lambda function to China regions, SAM could create a ECR repo, but failed to push the image to that repo. Below is the error message.

```shell
The push refers to repository [215043275130.dkr.ecr.cn-north-1.amazonaws.com/serverlesspetstore065dbd41/nginxfunctione76ecfa8repo]
Error: Unable to upload artifact nginxfunction:v1 referenced by ImageUri parameter of NginxFunction resource.
Get "https://215043275130.dkr.ecr.cn-north-1.amazonaws.com/v2/": Service Unavailable
```

The endpoint is wrong: it should be "https://215043275130.dkr.ecr.cn-north-1.amazonaws.com.cn". 

### Steps to reproduce:

1. create a image type lambda function with sam init
2. build the function with sam build
3. deploy the function to cn-north-1 with sam deploy -g. Let sam create create the ecr repo.

### Observed result:

```shell
sunhua@a07817ad97aa nginx % sam deploy -g --debug
2021-09-14 22:35:28,264 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2021-09-14 22:35:28,265 | Using config file: samconfig.toml, config environment: default
2021-09-14 22:35:28,265 | Expand command line arguments to:
2021-09-14 22:35:28,265 | --guided --template_file=/Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml --stack_name=sam-app --fail_on_empty_changeset 

Configuring SAM deploy
======================

        Looking for config file [samconfig.toml] :  Not found

        Setting default arguments for 'sam deploy'
        =========================================
        Stack Name [sam-app]: sam-petstore
        AWS Region [cn-north-1]: 
2021-09-14 22:35:33,946 | No Parameters detected in the template
2021-09-14 22:35:33,962 | 2 stacks found in the template
        #Shows you resources changes to be deployed and require a 'Y' to initiate deploy
        Confirm changes before deploy [y/N]: 
        #SAM needs permission to be able to create roles to connect to the resources in your template
        Allow SAM CLI IAM role creation [Y/n]: 
2021-09-14 22:35:43,052 | No Parameters detected in the template
2021-09-14 22:35:43,069 | 2 resources found in the stack 
2021-09-14 22:35:43,069 | No Parameters detected in the template
2021-09-14 22:35:43,082 | Found Serverless function with name='NginxFunction' and ImageUri='nginxfunction:v1'
2021-09-14 22:35:43,082 | --base-dir is not presented, adjusting uri ./app relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
2021-09-14 22:35:43,082 | --base-dir is not presented, adjusting uri . relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
2021-09-14 22:35:43,082 | No Parameters detected in the template
2021-09-14 22:35:43,093 | Detected Inline Swagger definition
2021-09-14 22:35:43,093 | Auth checks done on swagger are not exhaustive!
2021-09-14 22:35:43,093 | No Parameters detected in the template
2021-09-14 22:35:43,103 | Detected Inline Swagger definition
2021-09-14 22:35:43,103 | Auth checks done on swagger are not exhaustive!
        NginxFunction may not have authorization defined, Is this okay? [y/N]: y
        NginxFunction may not have authorization defined, Is this okay? [y/N]: y
2021-09-14 22:35:46,303 | No Parameters detected in the template
2021-09-14 22:35:46,334 | 2 resources found in the stack 
2021-09-14 22:35:46,334 | No Parameters detected in the template
2021-09-14 22:35:46,353 | Found Serverless function with name='NginxFunction' and ImageUri='nginxfunction:v1'
2021-09-14 22:35:46,353 | --base-dir is not presented, adjusting uri ./app relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
2021-09-14 22:35:46,353 | --base-dir is not presented, adjusting uri . relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
2021-09-14 22:35:46,353 | No function or layer definition found with code sign config, skipping
        Save arguments to configuration file [Y/n]: 
        SAM configuration file [samconfig.toml]: 
        SAM configuration environment [default]: 

        Looking for resources needed for deployment:
         Managed S3 bucket: aws-sam-cli-managed-default-samclisourcebucket-146gpe82azn5q
         A different default S3 bucket can be set in samconfig.toml
2021-09-14 22:35:53,935 | No Parameters detected in the template
2021-09-14 22:35:53,960 | 2 resources found in the stack 
2021-09-14 22:35:53,960 | No Parameters detected in the template
2021-09-14 22:35:53,979 | Found Serverless function with name='NginxFunction' and ImageUri='nginxfunction:v1'
2021-09-14 22:35:53,979 | --base-dir is not presented, adjusting uri ./app relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
2021-09-14 22:35:53,979 | --base-dir is not presented, adjusting uri . relative to /Users/sunhua/Projects/github.com/aws-samples/aws-lambda-adapter/examples/nginx/.aws-sam/build/template.yaml
         Image repositories: Not found.
         #Managed repositories will be deleted when their functions are removed from the template and deployed
         Create managed ECR repositories for all functions? [Y/n]: 

        Saved arguments to config file
        Running 'sam deploy' for future deployments will use the parameters saved above.
        The above parameters can be changed by modifying samconfig.toml
        Learn more about samconfig.toml syntax at 
        https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-config.html

The push refers to repository [215043275130.dkr.ecr.cn-north-1.amazonaws.com/sampetstore6514f79a/nginxfunctione76ecfa8repo]
2021-09-14 22:36:09,849 | Unable to export
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/packageable_resources.py", line 272, in export
    self.do_export(resource_id, resource_dict, parent_dir)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/packageable_resources.py", line 285, in do_export
    uploaded_url = upload_local_image_artifacts(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/utils.py", line 118, in upload_local_image_artifacts
    return uploader.upload(image_path, resource_id)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/ecr_uploader.py", line 86, in upload
    self._stream_progress(push_logs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/ecr_uploader.py", line 197, in _stream_progress
    self._stream_write(_id, status, progress, error)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.30.0/libexec/lib/python3.8/site-packages/samcli/lib/package/ecr_uploader.py", line 213, in _stream_write
    raise DockerPushFailedError(msg=error)
samcli.commands.package.exceptions.DockerPushFailedError: Get "https://215043275130.dkr.ecr.cn-north-1.amazonaws.com/v2/": Service Unavailable
2021-09-14 22:36:09,852 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '97c6bd28-bb3f-444b-bd63-3b646557274f', 'installationId': 'e1b63f42-409a-4c99-8247-834b2f55e71c', 'sessionId': 'd20e1521-51b1-4e73-8afd-65aa5ccdd6e8', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.8.12', 'samcliVersion': '1.30.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam deploy', 'duration': 41587, 'exitReason': 'ExportFailedError', 'exitCode': 1}}]}
2021-09-14 22:36:10,645 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)
Error: Unable to upload artifact nginxfunction:v1 referenced by ImageUri parameter of NginxFunction resource.
Get "https://215043275130.dkr.ecr.cn-north-1.amazonaws.com/v2/": Service Unavailable
```

### Expected result:

The function should be deployed without error. 

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS 
2. `sam --version`: SAM CLI, version 1.30.0
3. AWS region: cn-north-1

**Comment by CoshUS** [comment]
PR open for addressing this issue: https://github.com/aws/aws-sam-cli/pull/3280

**Comment by CoshUS** [comment]
Fixed in [SAM CLI 1.33.0 release](https://github.com/aws/aws-sam-cli/releases/tag/v1.33.0).
